subiquity/ui/views/filesystem/guided.py

```diff
@@ -173,34 +197,33 @@ def _toggle_guided(self, sender, new_value):
 """)
 
 
-class GuidedDiskSelectionView (BaseView):
+class GuidedDiskSelectionView(BaseView):
 
     title = _("Guided storage configuration")
 
-    def __init__(self, controller):
+    def __init__(self, controller, disks):
         self.controller = controller
 
-        found_disk = False
-        found_ok_disk = False
-        for disk in controller.model.all_disks():
-            found_disk = True
-            if disk.size > 6*(2**30):
-                found_ok_disk = True
-                break
-
-        if found_ok_disk:
-            self.form = GuidedForm(model=controller.model)
-
-            connect_signal(self.form, 'submit', self.done)
-            connect_signal(self.form, 'cancel', self.cancel)
-
-            super().__init__(
-                self.form.as_screen(focus_buttons=False, excerpt=_(subtitle)))
-        elif found_disk:
-            super().__init__(
-                screen(
-                    [Text(rewrap(_(no_big_disks)))],
-                    [other_btn(_("OK"), on_press=self.manual)]))
+        if disks:
+            found_ok_disk = False
+            for disk in disks:
+                if disk.size > 6*(2**30):
```

Comment:
Do we aspire that this code would be used in ubuntu core like cases where even smaller disks would be permissible?  At minimum I suggest that we pull this value out as a constant.
Further this could change over time.  I suppose it'd be nice to have the ISO carry an indicator of what the minimum should be, though maybe it changes infrequently enough that that is overkill and we can just bump it manually.

Reply:
Better calculation of the minimum disk space is on the desktop team's wish list already but yeah, having it as a duplicated magic constant here is silly. I'm also not sure why it's as high as this tbh. server needs / to be somewhere between 1 and 2 gigs currently, I guess desktop probably has slightly higher requirements. I moved the determination into the server -- at some point the server part of the installer will need to know if it is installing desktop or server or core and can use that to decide. But not today!